core/modules/node/tests/node.test

```diff
@@ -904,11 +904,10 @@ class NodeAccessRecordsUnitTest extends BackdropWebTestCase {
       $this->assertNotEqual($grants, $altered_grants, format_string('Altered the %op grant for a user.', array('%op' => $op)));
     }
 
-    // Check that core does not grant access to an unpublished node when an
-    // empty $grants array is returned.
+    // Check correct access is set for an unpublished node.
     $node6 = $this->backdropCreateNode(array('status' => 0, 'disable_node_access' => TRUE));
     $records = db_query('SELECT realm, gid FROM {node_access} WHERE nid = :nid', array(':nid' => $node6->nid))->fetchAll();
-    $this->assertEqual(count($records), 0, 'Returned no records for unpublished node.');
+    $this->assertEqual(count($records), 3, 'Returned no records for unpublished node.');
```

Comment:
`Returned no records for unpublished node.` is no longer accurate.

We need more tests here that actually return 0 rows in addition to these tests that return content. We need positive validation as well as negative validation.